hermes-consumers/src/main/java/pl/allegro/tech/hermes/consumers/consumer/result/MessageDeliveryHandler.java

```diff
@@ -0,0 +1,14 @@
+package pl.allegro.tech.hermes.consumers.consumer.result;
+
+import pl.allegro.tech.hermes.api.Subscription;
+import pl.allegro.tech.hermes.consumers.consumer.sender.MessageSendingResult;
+
+public interface MessageDeliveryHandler {
```

Comment:
I'm aware this interface (along with `MessageDeliveryHandlersController`) is sort of redundant, as we already have success and error handlers. `ConsumerMessageSender` should have access to a list of possible message result handlers (including these related to OAuth). `MessageDeliveryHandlersController` could be divided into message result success and error handlers + some kind of consumer lifecycle listeners (that listens to creation/shutdown of consumer, update of subscription). 
This looks like a bigger change for consumers module and we should consider doing this outside of this PR. So please, turn a blind eye, temporarily. 🙈